### PR TITLE
Communication Resource | API Version Update

### PR DESCRIPTION
**Pull request by mjafferi-msft**
# Description
Live test pipelines are failing due to the failure of deployment test resources in INT environment for all the modalities across all the platforms since the Communication Services API version : 2020-08-20-preview is obsolete, which was recently removed from support.
Updating API version to 2023-03-31.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/communication/test-resources/test-resources.json [hunk 1]
```diff
@@ -58,7 +58,7 @@
     },
     {
       "type": "Microsoft.Authorization/roleAssignments",
-      "apiVersion": "2019-04-01-preview",
+      "apiVersion": "2022-04-01",
```
Comment (AikoBB) [inline sdk/communication/test-resources/test-resources.json:61]: it should be update in other platforms as well?

File: sdk/communication/test-resources/test-resources.json [hunk 2]
```diff
@@ -82,15 +82,15 @@
     },
     "COMMUNICATION_LIVETEST_DYNAMIC_CONNECTION_STRING": {
       "type": "string",
-      "value": "[listKeys(resourceId('Microsoft.Communication/CommunicationServices',variables('uniqueSubDomainName')), '2021-10-01-preview').primaryConnectionString]"
+      "value": "[listKeys(resourceId('Microsoft.Communication/CommunicationServices',variables('uniqueSubDomainName')), '2023-03-31').primaryConnectionString]"
```
Comment (AikoBB) [inline sdk/communication/test-resources/test-resources.json:85]: the changes in 85 & 93th lines are not applied for other platforms, should we apply for others?

File: sdk/communication/test-resources/test-resources.json (hunk 1 above)
Comment (mjafferi-msft) [reply sdk/communication/test-resources/test-resources.json:61]: I had already done that and they are merged to main

File: sdk/communication/test-resources/test-resources.json (hunk 2 above)
Comment (mjafferi-msft) [reply sdk/communication/test-resources/test-resources.json:85]: I had already done that and they are merged to main

**Pull request merged**